### PR TITLE
Use DISCO APi for getting latest Mandrel releases

### PR DESCRIPTION
**Pull request by zakkak**
- Improve error log for wrong mandrel version and javaVersion comb.
- Use Foojay.io DISCO API to get latest Mandrel release
- Relax requirement of mandrel versions starting with `mandrel-`
- Restore windows latest test & add test for non-prefixed mandrel ver

Closes https://github.com/graalvm/setup-graalvm/issues/64

### CODE REVIEW

**Comment by zakkak** [comment]
@fniephaus I went with a mandrel specific patch for now, let me know if you would like to make it more generic so that we could use it for both mandrel and graal.

File: src/main.ts [hunk 1]
```diff
@@ -58,13 +58,7 @@ async function run(): Promise<void> {
           graalVMHome = await graalvm.setUpGraalVMJDKCE(javaVersion)
           break
         case c.DISTRIBUTION_MANDREL:
-          if (graalVMVersion.startsWith(c.MANDREL_NAMESPACE)) {
-            graalVMHome = await setUpMandrel(graalVMVersion, javaVersion)
-          } else {
-            throw new Error(
-              `Mandrel requires the 'version' option (see https://github.com/graalvm/setup-graalvm/tree/main#options).`
```
Comment (fniephaus) [inline src/main.ts:65]: You probably still want to throw this error if `graalVMVersion.length == 0`, no? If I read this correctly, `version: 'mandrel-22.2.0.0-Final'` and `version: '22.2.0.0-Final'` will now do the same thing, but `version` is still required.

File: src/mandrel.ts [hunk 2]
```diff
@@ -1,60 +1,142 @@
...
+  } catch (error) {
+    throw new Error(`Failed to extract tag from URI ${uri}: ${error}`)
+  }
+}
+
+export async function getLatestMandrelReleaseUrl(
+  javaVersion: string
+): Promise<string> {
+  const url = `${DISCO_API_BASE}?jdk_version=${javaVersion}&distribution=${c.DISTRIBUTION_MANDREL}&architecture=${c.JDK_ARCH}&operating_system=${c.JDK_PLATFORM}&latest=per_distro`
+  const _http = new httpClient.HttpClient('http-client-tests')
```
Comment (fniephaus) [inline src/mandrel.ts:62]: Not sure `http-client-tests` is supposed to be used in production code?

**Review by fniephaus — APPROVED** [review]
LGTM. I have two minor comments, but feel free to address them in a follow up PR.

**Comment by zakkak** [comment]
> LGTM. I have two minor comments, but feel free to address them in a follow up PR.

Thank you @fniephaus, I will.

**Pull request merged**